java/flight/flight-sql-jdbc-core/src/test/java/org/apache/arrow/driver/jdbc/utils/CoreMockedSqlProducers.java

```diff
@@ -324,12 +324,16 @@ public static void assertLegacyRegularSqlResultSet(
       actualHireDates.add((Date) resultSet.getObject(5));
       actualLastSales.add((Timestamp) resultSet.getObject(6));
     }
-    collector.checkThat(actualRowCount, is(equalTo(expectedRowCount)));
-    collector.checkThat(actualIds, is(expectedIds));
-    collector.checkThat(actualNames, is(expectedNames));
-    collector.checkThat(actualAges, is(expectedAges));
-    collector.checkThat(actualSalaries, is(expectedSalaries));
-    collector.checkThat(actualHireDates, is(expectedHireDates));
-    collector.checkThat(actualLastSales, is(expectedLastSales));
+
+    final int finalActualRowCount = actualRowCount;
```

Comment:
is this required to be final?

Reply:
I meant do we need a separate variable?

Reply:
If we don't use `final`, we'll see the error: "Variable used in lambda expression should be final or effectively final."